src/Layout/Layout.js

```diff
@@ -195,82 +223,140 @@ type Props = {
+    mdUpHidden,
+    lgUpHidden,
+    xlUpHidden,
+    xsDownHidden,
+    smDownHidden,
+    mdDownHidden,
+    lgDownHidden,
+    xlDownHidden,
+  };
+
+  // determine if we need to wrap with <Hidden/>
+  let isHidden = false;
+  forOwn(hiddenProps, (value) => {
+    if (typeof value !== 'undefined') {
+      isHidden = true;
+      return false; // break
+    }
+
+    return true;
+  });
+
   const classes = context.styleManager.render(styleSheet);
+  const className = classNames({
+    [classes.typeContainer]: container,
+    [classes.typeItem]: item,
+    [classes[`gutter-xs-${String(gutter)}`]]: container && gutter !== 0,
+    [classes[`direction-xs-${String(direction)}`]]: direction !== Layout.defaultProps.direction,
+    [classes[`wrap-xs-${String(wrap)}`]]: wrap !== Layout.defaultProps.wrap,
+    [classes[`align-xs-${String(align)}`]]: align !== Layout.defaultProps.align,
+    [classes[`justify-xs-${String(justify)}`]]: justify !== Layout.defaultProps.justify,
+    [classes['grid-xs']]: xs === true,
+    [classes[`grid-xs-${String(xs)}`]]: xs && xs !== true,
+    [classes['grid-sm']]: sm === true,
+    [classes[`grid-sm-${String(sm)}`]]: sm && sm !== true,
+    [classes['grid-md']]: md === true,
+    [classes[`grid-md-${String(md)}`]]: md && md !== true,
+    [classes['grid-lg']]: lg === true,
+    [classes[`grid-lg-${String(lg)}`]]: lg && lg !== true,
+    [classes['grid-xl']]: xl === true,
+    [classes[`grid-xl-${String(xl)}`]]: xl && xl !== true,
+  }, classNameProp);
+  const layoutProps = { className, ...other };
+
   // workaround: see https://github.com/facebook/flow/issues/1660#issuecomment-297775427
   const ComponentProp = component || Layout.defaultProps.component;
 
-  return (
-    <ComponentProp
-      className={classNames({
-        [classes.typeContainer]: container,
-        [classes.typeItem]: item,
-        [classes[`gutter-xs-${String(gutter)}`]]: container && gutter !== 0,
-        [classes[`direction-xs-${String(direction)}`]]: direction !== Layout.defaultProps.direction,
-        [classes[`wrap-xs-${String(wrap)}`]]: wrap !== Layout.defaultProps.wrap,
-        [classes[`align-xs-${String(align)}`]]: align !== Layout.defaultProps.align,
-        [classes[`justify-xs-${String(justify)}`]]: justify !== Layout.defaultProps.justify,
-        [classes['grid-xs']]: xs === true,
-        [classes[`grid-xs-${String(xs)}`]]: xs && xs !== true,
-        [classes['grid-sm']]: sm === true,
-        [classes[`grid-sm-${String(sm)}`]]: sm && sm !== true,
-        [classes['grid-md']]: md === true,
-        [classes[`grid-md-${String(md)}`]]: md && md !== true,
-        [classes['grid-lg']]: lg === true,
-        [classes[`grid-lg-${String(lg)}`]]: lg && lg !== true,
-        [classes['grid-xl']]: xl === true,
-        [classes[`grid-xl-${String(xl)}`]]: xl && xl !== true,
-      }, className)}
-      {...other}
-    >
-      {children}
-    </ComponentProp>
-  );
+  if (isHidden) {
+    return (
+      <Hidden {...hiddenProps}>
```

Comment:
That's a good argument. I wish we had a way to get some stats regarding who the components are used in the wild. We would make much better API choices. At least we have your data 👍 .